test/mockgateway/index.js

```diff
@@ -115,6 +127,15 @@ async function handleRequest (req) {
   return obj;
 }
 
-module.exports = function () {
-  return http.createServer(onReq);
+function validateHeader(txData) {
+  if (txData.startsWith('00736973002d00001{"expiry":12343333,"confidential":false}')) {
```

Comment:
TODO: make this validation/parsing more general once we finish the runtime side.

Reply:
should this be "if '!' startsWith"?

Reply:
Yes.

Reply:
Done. The tests were still passing because `0x` wasn't prefixed to the string.